EU4toV2/Source/V2World/Diplomacy/Diplomacy.cpp

```diff
@@ -218,6 +219,35 @@ void V2::Diplomacy::convertRelationsToInfluence(const std::map<std::string, std:
 	}
 }
 
+void V2::Diplomacy::sphereHRE(const EU4::World& sourceWorld, std::map<std::string, std::shared_ptr<Country>>& countries)
+{
+	if (const auto& hreReforms = sourceWorld.getHREReforms();
+		 std::find(hreReforms.begin(), hreReforms.end(), "emperor_reichskrieg") == hreReforms.end())
+	{
+		return;
+	}
+	else
+	{
+		Log(LogLevel::Info) << "\tSphereing HRE";
+
+		const auto& emperorItr = std::find_if(countries.begin(), countries.end(),
+			[](const std::pair<std::string, std::shared_ptr<Country>>& country)
```

Comment:
sending a reference to a shared pointer doesn't seem right. Add const.

Reply:
It's a modified copy of an std::find_if I used elsewhere. I'm not sure if I understand your suggestion. Could I just go with `[](std::pair<std::string, std::shared_ptr<Country>> country)`?

Reply:
It's a question what the lambda takes. Generally speaking:
const std::shared_ptr<Country>& - passes a const reference to a pointer to the object, which is rather simple and fast.
std::shared_ptr<Country>& - means this is a reference to the pointer that you're allowed to edit, which is absolutely wrong, if you edit the pointer it no longer points to the same place.
std::shared_ptr<Country> - passes a copy of the pointer to the object, and increases that shared pointer's internal counter. similar to the first case but a bit slower.

Reply:
If I understood correctly, lambda takes whatever is contained in the container I'm searching - in this case `std::map<std::string, std::shared_ptr<Country>>& countries`